models/agent_helpers.go

```diff
@@ -637,6 +637,63 @@ type CreateAgentParams struct {
 	DisableCollectors              []string
 }
 
+func verifyNodeAndAgent(nodeType NodeType, agentType AgentType) error {
+	switch nodeType {
+	case GenericNodeType:
+		return nil
+	case ContainerNodeType:
+		return nil
+	case RemoteNodeType:
+		if agentType == ExternalExporterType {
+			return nil
+		}
+	case RemoteRDSNodeType:
+		if agentType == RDSExporterType {
+			return nil
+		}
+	case RemoteAzureDatabaseNodeType:
+		if agentType == AzureDatabaseExporterType {
+			return nil
+		}
+	default:
+		return nil
+	}
+	return errors.Errorf("invalid combination of node type %s and agent type %s", nodeType, agentType)
+}
+
+func verifyServiceAndAgent(serviceType ServiceType, agentType AgentType) error {
+	switch agentType {
```

Comment:
maybe try to use a map for describing allowed relationships between agents and services? e.g.
```
m := map[AgentType]*ServiceType{
	PMMAgentType: nil,
	MySQLdExporterType: ServiceTypePointer(MySQLServiceType),
...
}
```